src/hooks/category.test.ts

```diff
@@ -0,0 +1,114 @@
+import nock from 'nock';
+import { List } from 'immutable';
+import { mockHook, setUpTest } from '../../test/utils';
+import { buildGetCategoriesRoute, buildGetCategoryInfoRoute, buildGetItemCategoriesRoute, buildGetItemsInCategoryRoute, GET_CATEGORY_TYPES_ROUTE } from '../api/routes';
+import { buildCategoriesKey, buildItemCategoryKey, buildItemsByCategoryKey, CATEGORY_KEY, CATEGORY_TYPES_KEY } from '../config/keys';
+import { CATEGORIES, CATEGORY_TYPES, ITEM_CATEGORIES } from '../../test/constants';
+import { Category, CategoryType, ItemCategory } from '../types';
+
+const { hooks, wrapper, queryClient } = setUpTest();
+
+type ItemId = {
+  item_id: string,
+}
+
+describe('Category Hooks', () => {
+  afterEach(() => {
+    nock.cleanAll();
+    queryClient.clear();
+  });
+
+  describe('useCategoryTypes', () => {
+    const route = `/${GET_CATEGORY_TYPES_ROUTE}`;
+    const key = CATEGORY_TYPES_KEY;
+
+    const hook = () => hooks.useCategoryTypes();
+
+    it(`Receive category types`, async () => {
+      const response = CATEGORY_TYPES;
+      const endpoints = [{ route, response }];
+      const { data } = await mockHook({ endpoints, hook, wrapper });
+
+      expect((data as List<CategoryType>).toJS()).toEqual(response);
+
+      // verify cache keys
+      expect(queryClient.getQueryData(key)).toEqual(List(response));
+    });
+  });
+
+  describe('useCategories', () => {
+    const typeIds = ['type-id1', 'type-id2'];
+    const route = `/${buildGetCategoriesRoute(typeIds)}`;
+    const key = buildCategoriesKey(typeIds);
+
+    const hook = () => hooks.useCategories(typeIds);
+
+    it(`Receive category types`, async () => {
+      const response = CATEGORIES;
+      const endpoints = [{ route, response }];
+      const { data } = await mockHook({ endpoints, hook, wrapper });
+
+      expect((data as List<Category>).toJS()).toEqual(response);
+
+      // verify cache keys
+      expect(queryClient.getQueryData(key)).toEqual(List(response));
+    });
+  });
+
+  describe('useCategory', () => {
+    const categoryId = 'id';
+    const route = `/${buildGetCategoryInfoRoute(categoryId)}`;
+    const key = CATEGORY_KEY;
```

Comment:
I think it's not good to use this key. This means that each time you fetch a category, it will be saved in the same key, even if the id is different. You should use a key depending on the category id.